docs/application/dotnet/guides/applications/nui-widget-app.md

```diff
@@ -0,0 +1,243 @@
+1. Define your widget application class, which is inherited from [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class:
+
+   ```csharp
+   class Program : NUIWidgetApplication
+   ```
+2. The widget application starts with `Main()`, which creates and initializes the application. `Run()` method of [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class is used to start the application event-loop.
+
+   `NUIWidgetApplication` class provides two kinds of constructors:
+
+   - For using the `NUIWidgetApplication(Type type)` constructor, in case the widget application's ID is the same as the application ID:
+      ```csharp
+        static void Main(string[] args)
+        {
+            var app = new Program(typeof(MyWidget));
+            app.Run(args);
+        }    
+      ```
+
+   - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo)` constructor, in case your widget applications have multiple widget classes.
+
+     For multiple instances, add `<widget-class>` in XML as follows:
+      ```XML
+          <widget-class classid="second" update-period="0">
+            <support-size preview="Widget.png">2x2</support-size>
+          </widget-class>
+      ```
+     You can then modify the code as follows:
+      ```csharp
+        static void Main(string[] args)
+        {
+           Dictionary<System.Type, string> widgetSet = new Dictionary<Type, string>();
+            widgetSet.Add(typeof(MyWidget), "second@org.tizen.example.WidgetTemplate");
+            var app = new Program(widgetSet);
+            app.Run(args);
+        }    
+      ```
+
+3. Define your widget class, which is inherited from [Widget](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Widget.html):
+
+   ```csharp
+   class MyWidget : Widget
+   ```
+
+   
+
+4. Override event callback methods of your new class:
+    ```csharp
+    class MyWidget : Tizen.NUI.Widget
+    {
+      protected override void OnCreate(string contentInfo, Window window)
+      {
+        /// Create the UI
+        /// ....
+        base.OnCreate(contentInfo, window);
+      }
+    }
+    ```
+
+   `Widget` class provides the following callback:
+
+   - `OnCreate()` callback is triggered when the widget instance is created.
+
+   - `OnTerminate()` callback is triggered when the widget instance is terminated:
+
+      ```csharp
+      protected override void OnTerminate(string contentInfo, TerminationType type)
+      ```
+
+   - `OnPause()` callback is triggered when the widget instance is paused.
+
+     Your application does the necessary actions when the widget instance becomes invisible because the Application framework can destroy a paused widget instance, for instance, such as the case that memory is lacked:
+
+      ```csharp
+      protected override void OnPause()
+      ```
+
+   - `OnResume()` callback is triggered when the widget instance is resumed.
```

Comment:
Please add a colon instead of a period at the end of the sentence.

```suggestion
- `OnResume()` callback is triggered when the widget instance is resumed:
```